Python/Product/Analysis/LanguageServer/CompletionAnalysis.cs

```diff
@@ -440,9 +440,17 @@ Func<TextReader> openDocument
         private IEnumerable<CompletionItem> GetCompletionsInForStatement() {
             if (Statement is ForStatement fs) {
                 if (fs.Left != null) {
-                    if (Index < fs.Left.StartIndex) {
-                        return null;
-                    } else if (Index <= fs.Left.EndIndex) {
+                    if (fs.InIndex > fs.StartIndex) {
+                        if (Index > fs.InIndex + 2) {
+                            return null;
+                        } else if (Index >= fs.InIndex) {
+                            ApplicableSpan = new SourceSpan(Tree.IndexToLocation(fs.InIndex), Tree.IndexToLocation(fs.InIndex + 2));
+                            return Once(InKeywordCompletion);
+                        }
+                    }
+                    if (fs.Left.StartIndex > fs.StartIndex && fs.Left.EndIndex > fs.Left.StartIndex && Index > fs.Left.EndIndex) {
+                        return Once(InKeywordCompletion);
+                    } else if (fs.ForIndex >= fs.StartIndex && Index > fs.ForIndex + 3) {
```

Comment:
`else` not needed

Reply:
But expresses the intent better. It's actively harmful to have an `else` on the previous one though (that bug took me a few minutes to find...)